### ISSUE TITLE
Is there any guide to choose the Algorithms

### ISSUE THREAD

**Issue by Edward-Zhou** [dotnet/machinelearning-samples#156]
For a new member to machinelearning, I found there are multiple algorithms for the same scenario like Regression, there are `Sdca Regression`, `FastTreeTweedie Regression`, `Fast Tree` and etc.      

How to choose which algorithms to use?

Or, we train the model with all the available algorithms and to choose the best one per to the Evaluate result.

Is there any doc to describe the algorithms, their advantages and disadvantages and when to use them?

**Comment by asthana86** [comment]
@glebuk @Zruty0 @JRAlexander 

Perhaps the folks that i added could help.

**Comment by glebuk** [comment]
Edward,
This choice truly the key question in getting good ML results!  The decision is based on many factors and is not something that can be simply guessed.  The selection depends on several criteria.  For example:
**Input training data**

- How large is the training data (rows / columns)
- What kind of distribution do you expect (different regressors are designed for different kinds of distribution)

**How will you use the model?**
- Speed and space requirenments
- Regulatory requirenments (does the model have to be explainable?)
- Accuracy requirenments - how important is it to get most accurate answer possible vs how robust you want the prediction to be?

The most simple practical approach, is to:
0. Ensure you properly featurize your data.
1. Try FastTree or LightGbm as it is usually good in most cases with default parameters.
2. Sweep hyper parameters of the FastTree
3. Try other algorithms and sweep them and compare results.
